pkg/network/state.go

```diff
@@ -913,6 +918,47 @@ func (ns *networkState) determineConnectionIntraHost(connections []ConnectionSta
 	}
 }
 
+// fixConnectionDirection fixes connection direction
+// for UDP incoming connections.
+//
+// Some UDP connections can be assigned an incoming
+// direction incorrectly since we cannot reliably
+// distinguish between a server and client for UDP
+// in eBPF. Both clients and servers can call
+// the system call bind() for source ports, but
+// UDP servers don't call listen() or accept()
+// like TCP.
+//
+// This function fixes only a very specific case:
+// incoming UDP connections, when the source
+// port is ephemeral but the destination port is not.
+// This is the only case where we can be sure the
+// connection has the incorrect direction of
+// incoming. For remote connections, only
+// destination ports < 1024 are considered
+// non-ephemeral.
+func fixConnectionDirection(c *ConnectionStats) {
```

Comment:
The backend gets the ephemeral port range for each host, could we move this logic into the backend so older agents pick this up/

Reply:
I don't think it does. We only indicate whether the the source port is ephemeral.

Reply:
Gah, what a bummer